### DISCUSSION TITLE
No "GraphQL Compiler" build option (Strawberry Shake)

### DISCUSSION THREAD

**Discussion by killnine** [ChilliCream/hotchocolate#6624] [Q&A]
Hey, all
I am setting up a strawberry shake client project and can't get it to build source. Documentation recommended changing the build action to 'GraphQL Compiler' but that option doesn't appear in VS or Rider for me.

![image](https://github.com/ChilliCream/graphql-platform/assets/660420/0a2dc164-77db-41ff-8359-e3a54aa49b5a)
![image (1)](https://github.com/ChilliCream/graphql-platform/assets/660420/6ccd65fe-fe4b-4825-a1af-8f3bba4592ed)

I have an `Infrastsructure` project to which I installed the StrawberryShake tooling, However, my app runs under a separate `Web` project. In any case, the project was able to get the `dotnet-tools.json` set up and I was able to init the schema from my existing GraphQL server endpoint.

But building wont generate code because I don't think it sees any GraphQL files (because they're not flagged to compile).

**Comment by killnine** [comment]
I think I may have figured it out from [this](https://stackoverflow.com/questions/76405803/c-sharp-net-build-failure-due-to-strawberryshake-error-method-not-found-voi) SO post.

So, I didn't have the `StrawberryShake.Server` nuget package. I don't see that being required in the [Getting Started](https://chillicream.com/docs/strawberryshake/v13/get-started) page at all...

Once I installed that in my `Infrastructure` project (its a class library, not an exe) it generated code.

**Reply by jzabroski** [reply]
That Getting Started page is for users who use a client store.  StrawberryShake.Server is for users who want a headless client (no UI) and just want the endpoint to be a console program or another backend server API consuming the GraphQL endpoint.  The terminology is slightly confusing and unique to GraphQL as far as I am aware, but the learning curve is shallow once you know the terms.
